### DISCUSSION TITLE
Setting the current working directory in launch.json

### DISCUSSION THREAD

**Discussion by sdegrande** [eclipse-cdt-cloud/cdt-cloud#33] [Q&A] [answered]
I can not find a way to set the cwd to be used during debug.
The "cwd" property in launch.json is not allowed...

**Comment by sdirix** [comment]
Hi @sdegrande,

What exactly are you trying to configure? CDT Cloud Blueprint contains a simple CMake example for which we also provide a (debug) launch configuration. For this simple scenario [it's enough to specify the program to debug](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/blob/4d0cc9e6be9bd1987719c060d751a8f137604242/theia-extensions/blueprint-example-generator/resources/cmake-example/.theia/launch.json).

To examine all configuration options for the CDT Cloud CDT GDB Adapter, see [here](https://github.com/eclipse-cdt-cloud/cdt-gdb-vscode/blob/4236db2900c8403be68dd4e37aed4f945fe4ee07/package.json#L59-L458).

**Reply by sdegrande** [reply]
As far as I can tell (by printing std::filesystem::current_path() in my code), when I start a debugging session, the cwd is my home dir (I'm running on Linux Fedora 38).
For some installation reasons, my program is looking for its asset files relatively to the cwd, and so it has to be ${workingFolder}.
With Eclipse a working directory can be defined in a debug configuration.
It seems to not (yet ?) be possible with cdt-cloud.

**Comment by jonahgraham** [comment, marked as answer]
There is an enhancement request to add support for current working directory of the inferior. Please see https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/issues/90

Patches welcome!

---

In the meantime you can add a custom command to [`initCommands`](https://github.com/eclipse-cdt-cloud/cdt-gdb-vscode/blob/4236db2900c8403be68dd4e37aed4f945fe4ee07/package.json#L108-L115) - perhaps something like `cd /path/to/desired/current/working/directory`

**Reply by sdegrande** [reply]
Works well, thanks !

**Reply by jonahgraham** [reply]
Great. Glad to hear it!
